src/panels/MissionsPanel/TimeLine/Rectangle.tsx

```diff
@@ -0,0 +1,80 @@
+import { Text, Tooltip } from '@mantine/core';
+import { ScaleLinear, ScaleTime } from 'd3';
+
+import { useOpenSpaceApi } from '@/api/hooks';
+import { useAppSelector } from '@/redux/hooks';
+import { DisplayType } from '@/types/enums';
+import { Phase } from '@/types/mission-types';
+
+import { DisplayedPhase } from '../MissionContent';
+import { jumpToTime } from '../util';
+
+interface RectangleProps {
+  scale: number; // d3 scale 'k' value
+  xScale: ScaleLinear<number, number, never>;
+  yScale: ScaleTime<number, number, never>;
+  nestedLevels: number;
+  setDisplayedPhase: (phase: DisplayedPhase) => void;
+  phase: Phase;
+  nestedLevel: number;
+  padding?: number;
+  color?: React.CSSProperties['color'];
+}
+
+export function Rectangle({
```

Comment:
Should this be called `PhaseRectangle` ? Since it is specifically used for phases?